pkg/terminal/command.go

```diff
@@ -1667,14 +1668,16 @@ func examineMemoryCmd(t *Term, ctx callContext, args string) error {
 				return fmt.Errorf("size must be a positive integer (<=8)")
 			}
 		default:
-			if i != len(v)-1 {
-				return fmt.Errorf("unknown option %q", v[i])
+			expr := strings.Join(v[i:], " ")
+			val, err := t.client.EvalVariable(ctx.Scope, expr, t.loadConfig())
+			if err != nil {
+				return err
 			}
-			// TODO, maybe we can support expression.
-			address, err = strconv.ParseUint(v[len(v)-1], 0, 64)
+			address, err = strconv.ParseUint(val.Value, 0, 64)
```

Comment:
I think this should take into account `val.RealType`, it's weird that `examinemem &x` wouldn't work.

Reply:
Good point 49cf77ca0f16909ad2c152ef8a4471c9e2ec4de0 tries to handle different return types. I ended up getting it to work without looking at the `val.RealType` field, hope that's okay.